lib/Common/Memory/RecyclerPointers.h

```diff
@@ -596,4 +598,35 @@ void *  __cdecl memset(_Out_writes_bytes_all_(_Size) WriteBarrierPtr<T> * _Dst,
     CompileAssert(false);
 }
 
-#include <Memory/WriteBarrierMacros.h>
\ No newline at end of file
+
+// This class abstract a pointer value with its last 2 bits set to avoid conservative GC tracking.
+template <class T>
+class GC_MARKED_OBJECT
```

Comment:
change the name to be taggedpointer? the mark has a strict definition in recycler which makes it confusing to use this name